content/chainguard/chainguard-images/getting-started/jre-minecraft/index.md

```diff
@@ -0,0 +1,556 @@
+
+## 4 – Persisting World Data
+
+So far, we’ve run a Minecraft server with an ephemeral world: anytime you remove the containers and recreate the environment, a new world is created, so you’ll lose any progress you have made, such as builds and achievements.
+
+Minecraft servers have a special directory where the world data is stored. We’ll need to set up a volume for persisting that data. This way, even if you remove all containers and rebuild your image, your world will be reinstated when you run `docker compose up` again, and you won’t lose your progress.
+
+If you haven’t yet, stop any running containers and bring your environment down:
+
+```shell
+docker compose down
+```
+
+Next, create a [named volume](https://docs.docker.com/engine/storage/volumes/#named-and-anonymous-volumes) to store your world data:
+
+```shell
+docker volume create guardcraft-world
+```
+
+This volume will be set in a location that can be obtained with a `docker volume inspect` command:
+
+```shell
+docker inspect guardcraft-world
+```
+
+```
+[
+    {
+        "CreatedAt": "2025-02-14T12:20:08Z",
+        "Driver": "local",
+        "Labels": null,
+        "Mountpoint": "/var/lib/docker/volumes/guardcraft-world/_data",
+        "Name": "guardcraft-world",
+        "Options": null,
+        "Scope": "local"
+    }
+]
+```
+
+The `Mountpoint` indicates where your world data will be persisted in the host machine.
+
+Next, open your `docker-compose.yaml` file using an editor of your choice and configure the `minecraft-java` service to use the named volume. You can replace the contents of the file with the following updated version:
+
+```yaml
+services:
+  minecraft-java:
+    image: cg-minecraft-server
+    build:
+      context: .
+    restart: unless-stopped
+    ports:
+      - 25565:25565
+    environment:
+      # Server properties Set Up
+      # MC_* variables will be replaced in the server.properties file
+      # Hyphens must be replaced with underscores
+      MC_gamemode: "survival"
+      MC_difficulty: "easy"
+      MC_motd: "Welcome to GuardCraft!"
+      MC_level_name: "GuardCraft"
+      MC_level_seed: "-3930782522362688416"
+    volumes:
+      - guardcraft-world:/usr/share/minecraft
+
+volumes:
+    guardcraft-world:
+      external: true
+```
+
+
+The `external: true` option tells Docker Compose that this volume is managed outside of the Compose lifecycle.
+
+You can now bring your environment up for the changes to take effect. You don’t need to rebuild the image, since there are no changes to the Dockerfile. Run:
+
+```shell
+docker compose up
+```
+
+This will start a new environment that uses the `guardcraft-world` volume to persist world data. If you check the volume mount now, it should be populated with the new world data:
```

Comment:
```suggestion
This will start a new environment that uses the `guardcraft-world` volume to persist world data. If you check the volume mount now, it will be populated with the new world data:
```